sample-plugin/pom.xml

```diff
@@ -39,6 +39,11 @@
                 <artifactId>json</artifactId>
                 <version>20230227</version>
             </dependency>
+            <dependency>
+                <groupId>org.jetbrains</groupId>
+                <artifactId>annotations</artifactId>
```

Comment:
@jtnord @MarkEWaite I thought the point of this update was to _remove_ all of these overrides?

Reply:
the kotlin override above (line 30) now be removed - alas the annotation is real conflict as it is bundled in multiple (two)  plugins (config as code and okhttp) and at a lower version on config-as-code.

Given the dependencies are listed alphabetically maven sees the dependency from c-as-c first and then gives you the lower bound.

I had thought about introducing a kotlin dependency - but a quick look seemed to show only okhttp requiring it, and introducing it just for a class retention annotation seemed like overkill.

Reply:
https://github.com/jenkinsci/okhttp-api-plugin/pull/142. Yet when they are removed:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (display-info) on project sample: 
[ERROR] Rule 5: org.apache.maven.enforcer.rules.dependency.RequireUpperBoundDeps failed with message:
[ERROR] Failed while enforcing RequireUpperBoundDeps. The error(s) are [
[ERROR] Require upper bound dependencies error for org.jetbrains:annotations:13.0 [test] paths to dependency are:
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.plugins:okhttp-api:4.10.0-127.vf711b_27a_f7fe [test]
[ERROR]     +-org.jetbrains:annotations:13.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.plugins:okhttp-api:4.10.0-127.vf711b_27a_f7fe [test]
[ERROR]     +-org.jetbrains.kotlin:kotlin-stdlib:1.8.10 [test]
[ERROR]       +-org.jetbrains:annotations:13.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-ast:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-builder:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-collection:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-data:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-dependency:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-format:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-html:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-misc:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-options:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-sequence:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-util-visitor:0.62.2 [test]
[ERROR]         +-org.jetbrains:annotations:15.0 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.vladsch.flexmark:flexmark-all:0.62.2 [test]
[ERROR]       +-com.vladsch.flexmark:flexmark-ext-abbreviation:0.62.2 [test]
[ERROR]         +-com.vladsch.flexmark:flexmark-util:0.62.2 [test]
[ERROR]           +-org.jetbrains:annotations:15.0 [test]
[ERROR] , 
[ERROR] Require upper bound dependencies error for org.json:json:20220320 [test] paths to dependency are:
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins:configuration-as-code:1569.vb_72405b_80249 [test]
[ERROR]     +-org.json:json:20220320 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-org.jenkins-ci.plugins:slack:629.vf00ea_cb_40d53 [test]
[ERROR]     +-org.json:json:20220924 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-io.jenkins.configuration-as-code:test-harness:1569.vb_72405b_80249 [test]
[ERROR]     +-com.github.erosb:everit-json-schema:1.14.1 [test]
[ERROR]       +-org.json:json:20220320 [test]
[ERROR] and
[ERROR] +-io.jenkins.tools.bom:sample:999999-SNAPSHOT
[ERROR]   +-org.jenkins-ci.plugins:jackson2-api:2.14.2-319.v37853346a_229 [test]
[ERROR]     +-com.fasterxml.jackson.datatype:jackson-datatype-json-org:2.14.2 [test]
[ERROR]       +-org.json:json:20190722 [test]
[ERROR] ]
```

`org.jetbrains:annotations` should I guess be excluded as a dep from both `okhttp-api` and `configuration-as-code`. `org.json:json` seems like an actual conflict (missing library plugin) between `configuration-as-code`, `slack`, and `jackson2-api`.

Reply:
> should I guess be excluded as a dep from ... configuration-as-code

Last time I did that it [broke config as code](https://github.com/jenkinsci/configuration-as-code-plugin/issues/1360) as it could no longer tell what was optional vs required by reading the various nullability annotations.

Granted this appears to be the test harness and flexmark which is *just* a markdown parser....

As it was an older version of an annotation I was concerned that the annotation could have changed in an incompatible way that would somehow break classloading.  (I know it is fine if missing - unless the annotation has RUNTIME retention - but was not sure if the annotation was present yet incompatible how things would fall out)

Reply:
Well the annotation dep could certainly be excluded from `okhttp-api`, right?

Reply:
what could possibly go wrong right :)

Reply:
https://github.com/jenkinsci/okhttp-api-plugin/pull/144